libyul/AsmAnalysis.cpp

```diff
@@ -702,6 +705,34 @@ bool AsmAnalyzer::validateInstructions(evmasm::Instruction _instr, SourceLocatio
 			"PC instruction is a low-level EVM feature. "
 			"Because of that PC is disallowed in strict assembly."
 		);
+	else if (m_eofVersion.has_value() &&
+				(
+					_instr == evmasm::Instruction::CALL ||
+					_instr == evmasm::Instruction::CALLCODE ||
+					_instr == evmasm::Instruction::DELEGATECALL ||
+					_instr == evmasm::Instruction::SELFDESTRUCT ||
+					_instr == evmasm::Instruction::JUMP ||
+					_instr == evmasm::Instruction::JUMPI ||
+					_instr == evmasm::Instruction::PC ||
+					_instr == evmasm::Instruction::CREATE ||
+					_instr == evmasm::Instruction::CODESIZE ||
+					_instr == evmasm::Instruction::CODECOPY ||
+					_instr == evmasm::Instruction::EXTCODESIZE ||
+					_instr == evmasm::Instruction::EXTCODECOPY ||
+					_instr == evmasm::Instruction::GAS
+				)
+			)
+	{
```

Comment:
It's easy to forget to handle something here, so I think it would be a good idea to add an extra assert at the end of this function, to check that the validation result (`true`/`false`) matches the value returned from `m_evmVersion.hasOpcode()`.

Reply:
Also, too much indent :)

```suggestion
	else if (m_eofVersion.has_value() && (
		_instr == evmasm::Instruction::CALL ||
		_instr == evmasm::Instruction::CALLCODE ||
		_instr == evmasm::Instruction::DELEGATECALL ||
		_instr == evmasm::Instruction::SELFDESTRUCT ||
		_instr == evmasm::Instruction::JUMP ||
		_instr == evmasm::Instruction::JUMPI ||
		_instr == evmasm::Instruction::PC ||
		_instr == evmasm::Instruction::CREATE ||
		_instr == evmasm::Instruction::CODESIZE ||
		_instr == evmasm::Instruction::CODECOPY ||
		_instr == evmasm::Instruction::EXTCODESIZE ||
		_instr == evmasm::Instruction::EXTCODECOPY ||
		_instr == evmasm::Instruction::GAS
	))
	{
```

Reply:
Fixed.

Reply:
Assert added but not comparing the return value of this function but rather `m_errorReporter.hasErrors()`